### PR TITLE
middleware: Add ShutdownBase and RebootBase RPCs

### PR DESCRIPTION
**Pull request by 0xB10C**
This PR implements following RPCs:

| RPC                | Input                 | Output                 |                  
|--------------------|-----------------------|-----------------------|
| ShutdownBase         | `dummyArg bool`  | ErrorResponse |
| RebootBase    | `dummyArg bool`  | ErrorResponse |

### CODE REVIEW

File: middleware/src/middleware.go [hunk 1]
```diff
@@ -440,6 +440,28 @@ func (middleware *Middleware) EnableClearnetIBD(enable bool) rpcmessages.ErrorRe
 	return rpcmessages.ErrorResponse{Success: true}
 }
 
+// ShutdownBase returns an ErrorResponse struct in response to a rpcserver request
+// It calls the bbb-cmd.sh script which initialtes a shutdown
```
Comment (Stadicus) [inline middleware/src/middleware.go]: s/initialtes/initializes/g

File: middleware/src/middleware.go [hunk 2]
```diff
@@ -440,6 +440,28 @@ func (middleware *Middleware) EnableClearnetIBD(enable bool) rpcmessages.ErrorRe
...
+	log.Println("shutting down the Base via the cmd script")
+	out, err := middleware.runBBBCmdScript("base", "shutdown", "")
+	if err != nil {
+		return rpcmessages.ErrorResponse{Success: false, Message: string(out), Code: err.Error()}
+	}
+	return rpcmessages.ErrorResponse{Success: true}
+}
+
+// RebootBase returns an ErrorResponse struct in response to a rpcserver request
+// It calls the bbb-cmd.sh script which initialtes a reboot
```
Comment (Stadicus) [inline middleware/src/middleware.go]: s/initialtes/initializes/g

File: middleware/src/middleware_test.go [hunk 3]
```diff
@@ -351,3 +351,21 @@ func TestSetHostname(t *testing.T) {
 	require.Equal(t, false, repsonse7.Success)
```
Comment (Stadicus) [inline middleware/src/middleware_test.go]: s/repsonse7/response7/g (multiple occurences)

**Comment by 0xB10C** [comment]
Addressed nits

**Pull request merged**